src/Panel/Ui/FilePreview/ImageFilePreview.php

```diff
@@ -30,17 +31,22 @@ public static function accepts(File $file): bool
 
 	public function details(): array
 	{
-		return [
-			...parent::details(),
-			[
+		$details    = parent::details();
+		$dimensions = $this->file->dimensions();
+
+		if ($dimensions instanceof Dimensions) {
```

Comment:
`Question`: How can `$file->dimensions()` ever not return a `Dimensions` object? As far as I can tell, `File` proxies the call to `Image`, which has a `dimensions()` method that is type-hinted to always return `Dimensions`.

Also I don't really understand how this change relates to the field methods refactoring.

Reply:
I was thinking that this is needed for all non-image files because then `$file->dimensions()` would return a field instance and with his PR non existent field method calls throw an exception now.

But then again, you are right, this is the image file preview. The file should always be an image.

I realized that `ImageFilePreviewTest::testProps()` seems to be wrongly implemented and threw me off in that direction.

Reply:
Ah, that makes sense. Thanks for the explanation :)